cupy/cuda/stream.pyx

```diff
@@ -323,3 +333,4 @@ class ExternalStream(BaseStream):
 
 
 Stream.null = Stream(null=True)
+Stream.ptds = Stream(ptds=True)
```

Comment:
That would work too, but I'm not sure if it's the right solution for two reasons:

1. The null stream may become ambiguous, are we really using the null stream or the per-thread one?
1. It's not anymore possible to enforce an explicit usage of the default stream when the environment variable is on, not sure how useful it is to use the default stream in that case anyway but it's probably useful to have that option.

Reply:
Yeah it's best for the core devs to decide, as I could be overlooking some design issues. My thoughts are:
1. `Stream.null` in CuPy means the default stream, and previously it meant the legacy one.
2. If a user explicitly asks for PTDS by setting the env var, it implicitly suggests that the legacy default stream should be voided to the maximum possible extent.

Reply:
I agree that the legacy stream should be avoided when PTDS is requested, but forcing that the `Stream.null` is indeed the PTDS stream will leave no choice for the user to fallback to the legacy one if necessary for some reason, so perhaps it's good to still have that possibility regardless.

Reply:
@pentschev I just noticed in CuPy's test suite it's expected that when users do not specify any stream, this is true:
```python
assert cuda.Stream.null == cuda.get_current_stream()
```
See https://github.com/cupy/cupy/blob/master/tests/cupy_tests/cuda_tests/test_stream.py. I guess this assumption might have propagated to some user codes as well? At least I just recalled I got one 😅 But we don't plan to use PTDS so it's probably OK.

Reply:
Hmm, that's interesting, indeed I didn't see that before. I suppose this could pose a problem, so we should probably try to disambiguate that first. I would still prefer to have a possibility to enforce using `Stream.null`, even when PTDS is enabled, but I'm happy to change that if the preferred is not to.

Reply:
Is using `Stream.null` when PTDS is enabled allowed?

Reply:
I think it is, but the semantics needs to be clearly defined (which is what we were discussing above -- should Stream.null mean stream 1 (legacy) or 2 (per-thread) when it's enabled?)

Reply:
We discussed and the majority was in favor of leaving `Stream.null` as the legacy one.
Anyway, we can just proceed as it is now and revisit it later.

Reply:
Thanks for discussing that!